### PR TITLE
[#1571][FOLLOWUP] improvement(server): Fail fast when the shuffleId is cleaned up

### PR DESCRIPTION
**Pull request by rickyma**
### What changes were proposed in this pull request?

Fail fast when the `shuffleId` is cleaned up.

### Why are the changes needed?

A follow-up PR for: https://github.com/apache/incubator-uniffle/pull/1574.

I encountered a scenario where, during the process of cleaning up the `appId`, if it is a high-concurrency and high-pressure situation, there will still be a large number of requests in flight to cache shuffle data sent to the server in a short period of time. Due to [#1574](https://github.com/apache/incubator-uniffle/pull/1574), the requests are waiting for the lock to be released at the `flushBuffer` method, which may cause the accumulation of pre-allocated size and `usedSize`. 
I think this can be optimized.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing UTs.

### CODE REVIEW

File: server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java [hunk 1]
```diff
@@ -268,6 +270,20 @@ public void sendShuffleData(
                 + spd.getPartitionId()
                 + "]";
         try {
+          isShuffleIdDeleted =
```
Comment (jerqi) [inline server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java:273]: what will happen it if server  processes a new application?

Comment (rickyma) [reply server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java:273]: I think a new `appId` and `shuffleId` will be already registered during the `registerShuffle` phase, which means `isShuffleIdDeleted` will be false when a new application arrives.

**Comment by jerqi** [comment]
Could you add a simple ut for this case?

**Comment by rickyma** [comment]
> Could you add a simple ut for this case?

I don't think we need extra UTs for this? This is an improvement. We simply want it to fail fast in advance. It doesn't affect the main process, but merely avoids executing unnecessary steps. What UTs would you like to be added? I have no idea.

**Comment by rickyma** [comment]
After rethinking about this, I will close this PR first.

**Pull request closed without merging**